pkg/catalog/catalog.go

```diff
@@ -1380,7 +1381,7 @@ func listCommitsWithoutPaths(it graveler.CommitIterator, params LogParams) ([]*C
 
 func foundAllCommits(params LogParams, commits []*CommitLog) bool {
 	return (params.Limit && len(commits) >= params.Amount) ||
-		len(commits) >= params.Amount+1
+		len(commits) >= params.Amount+1 || (len(commits) > 0 && commits[len(commits)-1].Reference == params.StopAt)
```

Comment:
@idanovo usually for references we call dereference before we use them. it means that we can accept tag, short sha or even branch to dereference the commit id.
From what I see so far we just use it as raw string all the way to here.